src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx

```diff
@@ -223,32 +225,45 @@ export const useGetTradeQuotes = () => {
       )
 
     return hopExecutionMetadata?.state === HopExecutionState.AwaitingSwap
-  }, [hopExecutionMetadata?.permit2, hopExecutionMetadata?.state])
-
-  const shouldFetchTradeQuotes = useMemo(
-    () =>
-      Boolean(
-        hasFocus &&
-          // Only fetch quote if the current "quote" is a rate (which we have gotten from input step)
-          activeTrade &&
-          !isExecutableTradeQuote(activeTrade) &&
-          // and if we're actually at pre-execution time
-          isFetchStep &&
-          sellAccountId &&
-          sellAccountMetadata &&
-          receiveAddress,
-      ),
-    [hasFocus, activeTrade, isFetchStep, sellAccountId, sellAccountMetadata, receiveAddress],
-  )
+  }, [hopExecutionMetadata?.permit2, hopExecutionMetadata?.state, swapperName])
+
+  const shouldFetchTradeQuotes = useMemo(() => {
+    // This isn't a mistake, we're not fetching anything.
+    // Li.Fi rate is an actual quote and we want to leverage cache we got from "rate" time, see swapperApi's `getTradeQuote` for more details
```

Comment:
Pretty much the secret sauce here: the fetch is not a fetch, since we're not using `forceRefetch: true`, so this really means `shouldLeverageCachedRTkQueryMetaStateForLiFi` but that may have been a tad bit verbose.